lib/fileset/internal.nix

```diff
@@ -461,6 +461,44 @@ rec {
     else
       nonEmpty;
 
+  # Turn a builtins.filterSource-based source filter on a root path into a file set containing only files included by the filter
+  # Type: Path -> (String -> String -> Bool) -> fileset
+  _fromSource = root: filter:
```

Comment:
`predicate` makes it a bit harder to understand, because it's really the `filter` argument to `builtins.path`/`builtins.filterSource`.

I now changed this to `sourceFilter` though, which both gets across that it's the source filter, and that it's not the same as `lib.filter`.

Reply:
That's an unfortunate name to have to inherit, but you've found a good solution.